rust/parquet/src/arrow/record_reader.rs

```diff
@@ -424,13 +389,16 @@ impl<T: DataType> RecordReader<T> {
     /// Split values into records according repetition definition and returns number of
     /// records read.
     fn split_records(&mut self, records_to_read: usize) -> Result<usize> {
-        let rep_levels_buf = self
+        let rep_levels = self
             .rep_levels
             .as_mut()
-            .map(|buf| FatPtr::<i16>::with_offset(buf, 0));
-        let rep_levels_buf = rep_levels_buf.as_ref().map(|x| x.to_slice());
+            .map(|buf| {
+                let (prefix, rep_levels, suffix) = unsafe { buf.data_mut().align_to_mut::<i16>() };
```

Comment:
It looks like this doesn't need to be mutable. Compiles fine if I change to the below (and changing the `as_mut` to `as_ref`
```suggestion
                let (prefix, rep_levels, suffix) = unsafe { buf.data().align_to::<i16>() };
```